autogluon/utils/tabular/ml/models/knn/knn_utils.py

```diff
@@ -53,6 +53,8 @@ def __init__(self, n_neighbors = 5, weights='uniform', n_jobs = -1, index_factor
 
         The model itself is a clone of the sklearn one
         """
+        try_import_faiss()
+        import faiss
```

Comment:
You can just set `self.faiss = faiss` and then you don't need to import it in every method